tools/heir-opt.cpp

```diff
@@ -102,6 +104,15 @@ int main(int argc, char **argv) {
   // Custom passes in HEIR
   mlir::heir::poly::registerPolyToStandardPasses();
   mlir::heir::bgv::registerBGVToPolyPasses();
+  // Register yosys optimizer pipeline
+  const char *abcEnvPath = std::getenv("HEIR_ABC_PATH");
+  const char *yosysRunfilesEnvPath = std::getenv("HEIR_YOSYS_LIBS");
+  if (abcEnvPath == nullptr || yosysRunfilesEnvPath == nullptr) {
+    llvm::errs() << "external files for yosys optimizer not found, please set "
+                    "HEIR_ABC_PATH and HEIR_YOSYS_LIBS\n";
+    return EXIT_FAILURE;
+  }
```

Comment:
I debated placing this inside of the reigstration function and having the register func instead return a logical result...

Reply:
Added some docs about disabling the build/test yosys dependent targets, and also included added documentation on the yosys-optimizer pass env var requirements and how to disable the pass.

Reply:
This also fixes https://github.com/google/heir/issues/118 with a test tag filter & build setting now.